Add Dark theme support

prupani-7 opened the pull request:
## Description

PR to add Add dark/light mode support for Kotlin samples

## Links and Data

Sample Epic: `runtime/kotlin/issues/1949`

## What To Review

-  Review the style/color/layout files for light and dark theme.

## How to Test

Run each sample to see the desired theme change in light and dark mode

<!-- OPTIONAL
## To Discuss
-->

<!-- OPTIONAL
## Screenshots
-->

shubham7109 reviewed (commented):
Reviewing selected samples ...

prupani-7 commented:
@shubham7109 The changes done for material button and alertdialog seem to be working as expected.

Concern: for find-route-in-transport-network sample, the `clearButton` is a MaterialButton which has an icon and text to it.
The icon cant be seen when the button is enabled. I think best is to remove the icon and just keep the text? Let me know what you think.

shubham7109 commented on samples-lib/src/main/res/values/styles.xml:
```diff
@@ -0,0 +1,39 @@
+    </style>
+
+    <style name="RegularButton" parent="@style/Widget.MaterialComponents.Button.TextButton">
+        <item name="android:textColor">@color/textbutton_textcolor</item>
```
```suggestion
        <item name="android:textColor">@color/textbutton_textcolor</item>
        <item name="iconTint">@color/textbutton_textcolor</item>
```
You can set the icon tint color to match the same as the button's text color.

prupani-7 commented:
Issue: The cut-Geometry sample, uses MaterialButtons. I am not able to see the buttons in both mode. Investigating..

In some samples, like `create convex hull` or `apply function to raster from service` where reset button is used, the Camel casing looks disabled for some reason. This used to be fine before, no changes have been made in this front.

shubham7109 commented on download-vector-tiles-to-local-cache/src/main/java/com/esri/arcgismaps/sample/downloadvectortilestolocalcache/MainActivity.kt:
```diff
@@ -45,14 +43,15 @@ import com.arcgismaps.tasks.exportvectortiles.ExportVectorTilesResult
 
     private val downloadArea: Graphic = Graphic()
-    private var dialog: AlertDialog? = null
+    private var dialog: androidx.appcompat.app.AlertDialog? = null
```
Can you add the import statement here 
```kotlin
import androidx.appcompat.app.AlertDialog
```
```suggestion
    private var dialog: AlertDialog? = null
```

prupani-7 replied:
Sorry i thought I changed it, to add the import and not use it that way.

shubham7109 commented on download-vector-tiles-to-local-cache/src/main/java/com/esri/arcgismaps/sample/downloadvectortilestolocalcache/MainActivity.kt:
```diff
@@ -255,7 +254,7 @@ class MainActivity : AppCompatActivity() {
     private fun createProgressDialog(exportVectorTilesJob: ExportVectorTilesJob): ProgressDialogLayoutBinding {
         val dialogLayoutBinding = ProgressDialogLayoutBinding.inflate(layoutInflater)
-        val dialogBuilder = AlertDialog.Builder(this@MainActivity).apply {
+        val dialogBuilder = androidx.appcompat.app.AlertDialog.Builder(this@MainActivity).apply {
```
```suggestion
        val dialogBuilder = AlertDialog.Builder(this@MainActivity).apply {
```

shubham7109 reviewed (commented):
Amazing job with this PR  @prupani-7  :tada: 

Just some minor comments here.

prupani-7 commented:
@shubham7109 I updated to use the import androidx.appcompat.app.AlertDialog

shubham7109 commented:
Tested all the samples using the SV. Just two minor comments: 

- Contingent values: The bottom sheet top bar is not purple. If that is a intentional change, update the "Apply" and "Cancel" buttons of the bottom sheet to use the outline style. 
- Analyze network with subnetwork trace: I think the original sample had a set background for the "Barrier conditions" that are added. Can it be supported in Dark mode?

prupani-7 commented:
> * Contingent values: The bottom sheet top bar is not purple. If that is a intentional change, update the "Apply" and "Cancel" buttons of the bottom sheet to use the outline style.

That's default from the ScrollView that we use. I could update the buttons to use Outlined style in this case.

prupani-7 commented:
> * Contingent values: The bottom sheet top bar is not purple. If that is a intentional change, update the "Apply" and "Cancel" buttons of the bottom sheet to use the outline style.

Interesting! It's actually a default from the ScrollView. And the `Apply` and `Cancel` are TextViews and not Buttons. Though they do behave like buttons.

prupani-7 commented:
> * Analyze network with subnetwork trace: I think the original sample had a set background for the "Barrier conditions" that are added. Can it be supported in Dark mode?

Yes, it can be supported. Not 100% sure of they way it looks in Dark mode, so removed the background set on it.
` android:background="@drawable/border" `
<img width="386" alt="image" src="https://user-images.githubusercontent.com/44178002/230136172-0405a283-3d91-4850-88ca-26aaa82adf90.png">

prupani-7 commented:
> * Contingent values: The bottom sheet top bar is not purple. If that is a intentional change, update the "Apply" and "Cancel" buttons of the bottom sheet to use the outline style.

Addressed the feedback

shubham7109 commented on analyze-network-with-subnetwork-trace/src/main/res/layout/activity_main.xml:
```diff
@@ -277,7 +277,7 @@
                     android:layout_marginTop="8dp"
                     android:layout_marginEnd="16dp"
                     android:layout_marginBottom="8dp"
-                    android:background="@drawable/border"
```
This line could be added back but will need to remove the solid background from the drawable. 
```xml
<solid android:color="#A9E8E8E8" />
```

shubham7109 commented on add-features-with-contingent-values/src/main/res/layout/add_feature_layout.xml:
```diff
@@ -26,34 +25,33 @@
-                <android.widget.TextView
+                <com.google.android.material.button.MaterialButton
                     android:id="@+id/applyTv"
+                    style="?attr/materialButtonOutlinedStyle"
```
Use custom style, instead of default style. (line 37 & 49)

```suggestion
                    style="@style/OutlineButton"
```

prupani-7 replied:
I think we should use the default style here, as it will ultimately use the custom style per the samples lib styles.xml
Else we might end up updating at all places in the samples(for consistency), which use the default style to use the custom style in their layout. 
and the style file change ` <item name="materialButtonOutlinedStyle">@style/OutlineButton</item>` would be redundant?

shubham7109 reviewed (changes requested):
Hopefully, this feedback helps in how to modify the two samples.

shubham7109 commented on samples-lib/src/main/res/values/styles.xml:
```diff
@@ -0,0 +1,45 @@
+        <item name="iconTint">@color/textbutton_textcolor</item>
+    </style>
+
+    <style name="TextInputLayoutTheme" parent="@style/Widget.MaterialComponents.TextInputLayout.FilledBox.ExposedDropdownMenu">
+        <item name="boxStrokeColor">#00FFFFFF</item>
+    </style>
```
Could you help me understand the reason for this custom theme? Seems like it is only used in one sample, and I find it helpful to have a stroke color to know which view is currently in focus. 

Would prefer if this could be removed.

prupani-7 replied:
I havn't added it. Just moved this theme from `add-features-with-contingent-values/styles.xml` to `samples-lib/styles.xml` since we don't have the styles for individual samples anymore.

prupani-7 replied:
I believe you worked on this sample? `add-features-with-contingent-values`
do you want me to remove it? its being used in `add-feature-layout.xml` file to set the style for `TextInputLayout`

shubham7109 replied:
Remove it 👍

prupani-7 replied:
Removed the custom style and added `style="@style/Widget.MaterialComponents.TextInputLayout.FilledBox.ExposedDropdownMenu"` to the textInputLayout's in respective xml of the sample.

shubham7109 reviewed (approved):
Tested all the samples using the SV. 

LGTM 👍

prupani-7 commented:
The latest sample viewer apk for testing can be found here -> smb://shortbread/public/PriyankaRupani/_sample-viewer-app-debug.apk

kaushikrw reviewed (approved):
@prupani-7 Looks great!
